src/averageAges.js

```diff
@@ -55,7 +71,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people.filter(({ mother, sex }) => {
+    return onlyWithSon
+      ? sex === 'm' && people.find((person) => person.name === mother)
+      : people.find((person) => person.name === mother);
+  });
+
+  const ages = mothers.reduce((acc, { mother, born: ageChild }) => {
+    const { born: ageMom } = people.find(({ name }) => {
+      return name === mother;
+    });
+    const diff = ageChild - ageMom;
+
+    return acc + diff;
+  }, 0);
+
+  return ages / mothers.length;
```

Comment:
Good job.
I just want to show example, how we can avoid O**2 complexity, and solve it just in two iterations

```
const mothers = people.reduce((acc, person) => {
    acc[person.name] = person;

    return acc;
  }, {});

  const { sum, childrenCount } = people.reduce((acc, person) => {
    const { sex, mother, born } = person;

    const isPerson = onlyWithSon
      ? sex === 'm' && mothers[mother]
      : mothers[mother];

    return isPerson ? {
      sum: acc.sum + born - mothers[mother].born,
      childrenCount: acc.childrenCount + 1,
    } : acc;
  }, {
    sum: 0, childrenCount: 0,
  });

  return sum / childrenCount;